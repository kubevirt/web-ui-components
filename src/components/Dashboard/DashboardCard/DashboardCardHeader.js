import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { CardHeading } from 'patternfly-react';

const DashboardCardHeader = ({ className, children, ...props }) => (
  <CardHeading {...props} className={classNames('kubevirt-dashboard__card-heading', className)}>
    {children}
  </CardHeading>
);

DashboardCardHeader.defaultProps = {
  className: null,
  children: null,
};

DashboardCardHeader.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
};

export default DashboardCardHeader;
