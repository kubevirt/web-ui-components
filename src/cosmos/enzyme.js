import createContext from 'react-cosmos-test/enzyme';

import paths from '../../config/paths';

export default function createTestContext(args) {
  args.proxies = args.proxies || []; // workaround of react-cosmos-test bug
  return createContext({
    ...args,
    cosmosConfigPath: `${paths.config}/cosmos.config.js`,
  });
}
