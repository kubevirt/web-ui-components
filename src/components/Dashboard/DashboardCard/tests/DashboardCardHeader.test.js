import React from 'react';
import { shallow } from 'enzyme';

import DashboardCardHeader from '../DashboardCardHeader';

const testDashboardCardHeader = () => <DashboardCardHeader>content</DashboardCardHeader>;

describe('<DashboardCardHeader />', () => {
  it('renders correctly', () => {
    const component = shallow(testDashboardCardHeader());
    expect(component).toMatchSnapshot();
  });
});
