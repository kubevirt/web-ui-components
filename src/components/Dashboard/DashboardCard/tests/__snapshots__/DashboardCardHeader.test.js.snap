// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardCardHeader /> renders correctly 1`] = `
<CardHeading
  className="kubevirt-dashboard__card-heading"
>
  content
</CardHeading>
`;
